llm-lora-finetuning/pipelines/train.py

```diff
@@ -0,0 +1,70 @@
+# Apache Software License 2.0
+#
+# Copyright (c) ZenML GmbH 2024. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+
+from steps import configure, evaluate_model, finetune, prepare_data, promote
+from zenml import logging as zenml_logging
+from zenml import pipeline
+from zenml.config import DockerSettings
+
+zenml_logging.STEP_LOGS_STORAGE_MAX_MESSAGES = (
+    10000  # workaround for https://github.com/zenml-io/zenml/issues/2252
+)
+
+
+@pipeline(
+    settings={
+        "docker": DockerSettings(
+            parent_image="pytorch/pytorch:2.2.2-cuda11.8-cudnn8-runtime",
+            requirements="requirements.txt",
+        )
+    }
+)
+def llm_peft_full_finetune():
+    """Pipeline for finetuning an LLM with peft.
+    
+    It will run the following steps:
+
+    - configure: set the system prompt and base model id
+    - prepare_data: prepare the datasets and tokenize them
+    - finetune: finetune the model
+    - evaluate_model: evaluate the base and finetuned model
+    - promote: promote the model to the target stage, if evaluation was successful
+    """
+    system_prompt, base_model_id = configure()
```

Comment:
I'm wondering, why have these as a separate step and not as pipeline parameters? So we store them in the artifact store for visibility?

Reply:
Moved it to logging in a form of metadata